server/concepts/Tournament.ts

```diff
@@ -81,19 +96,24 @@ export default class Tournament implements TournamentWihtoutMS {
         a: data.teamA,
         b: data.teamB,
       };
-      if (match.score.reportedByA === match.score.reportedByB) match.isFinished = true;
     } else if (assignedTeam === 'teamB') {
       if (match.score.reportedByB.a !== -1) throw new AppError('The match result has already been reported', 400);
       match.score.reportedByB = {
         a: data.teamA,
         b: data.teamB,
       };
-      if (match.score.reportedByA === match.score.reportedByB) match.isFinished = true;
     } else {
       throw new AppError('You are not authorized to update this match', 403);
     }
-
-    await TournamentRepository.updateMatch(matchId, match);
+    // If both reported scores are the same, && both teams reported scores
+    if (match.score.reportedByA === match.score.reportedByB && match.score.reportedByA.a !== -1) {
+      match.score.final = {
+        a: match.score.reportedByA.a,
+        b: match.score.reportedByA.b,
+      };
+      match.isFinished = true;
+    }
```

Comment:
It's not good. Bcz when owner want to overwrite scores, this cod/if will overwrite again

Reply:
It will not override in any case

Reply:
Only one case when this code will override score updated by owner is when owner will try to update score which was already updated by both teams, and is the same. Do we want to allow owner to override score after it is finished?

Reply:
I thought that owner would have only one chance to set final score, now owner can always set scores.